### PR TITLE
samples: drivers: fpga: merge both samples and add sample.yaml

### LINKED ISSUE #47613: Samples / Tests without a testcase.yaml or sample.yaml
**Describe the bug**

The following samples and tests do no have an meta data to allow them to be run via twister:

A few bsim tests were also found, but those are being run differently.

- tests/cmake/config_dir/foo does not have a sample.yaml or testcase.yaml
- tests/arch/xtensa_asm2 does not have a sample.yaml or testcase.yaml
- samples/boards/nrf/nrf53_sync_rtc/net does not have a sample.yaml or testcase.yaml
- samples/drivers/mbox/remote does not have a sample.yaml or testcase.yaml (@carlocaione)
- samples/drivers/fpga/fpga_controller_shell does not have a sample.yaml or testcase.yaml
- samples/drivers/fpga/fpga_controller does not have a sample.yaml or testcase.yaml
- samples/drivers/ipm/ipm_mcux/remote does not have a sample.yaml or testcase.yaml
- samples/subsys/ipc/openamp/remote does not have a sample.yaml or testcase.yaml
- samples/subsys/ipc/rpmsg_service/remote does not have a sample.yaml or testcase.yaml
- samples/subsys/ipc/ipc_service/icmsg/remote does not have a sample.yaml or testcase.yaml (@carlocaione)
- samples/subsys/ipc/ipc_service/static_vrings/remote does not have a sample.yaml or testcase.yaml (@carlocaione)
- samples/subsys/ipc/openamp_performance/remote does not have a sample.yaml or testcase.yaml (@carlocaione)
- samples/subsys/shell/devmem_load does not have a sample.yaml or testcase.yaml

**To Reproduce**

Looks a the directories above and find that there are no yaml files.

**Expected behavior**
All code in the tree needs to be tested, or at least built, to avoid bitrot. 

**Impact**
Bitrot.

### PR DESCRIPTION
**Pull request by aasthagr**
Merge the fpga_controller and fpga_controller shell
sample applications and add sample.yaml to enable testing in
twister.

Partial fix for #47613

Signed-off-by: Aastha Grover <aastha.grover@intel.com>

### CODE REVIEW

File: samples/drivers/fpga/fpga_controller/README.md [hunk 1]
```diff
@@ -1,10 +1,12 @@
 # Zephyr FPGA controller
```
[the remark this answers is not in the document]
Comment (aasthagr) [reply samples/drivers/fpga/fpga_controller/README.md]: Done. Thanks for noticing it.

File: samples/drivers/fpga/fpga_controller/src/main.c [hunk 2]
```diff
@@ -23,6 +24,14 @@ void main(void)
 			       PAD_SMT_DISABLE | PAD_REN_DISABLE | PAD_SR_SLOW |
 			       PAD_CTRL_SEL_AO_REG); /* Enable green led */
 
+#if CONFIG_SHELL
+	printk("Address of the bitstream (red): %p\n", &axFPGABitStream_red);
+	printk("Address of the bitstream (green): %p\n", &axFPGABitStream_green);
+	printk("Size of the bitstream (red): %d\n", sizeof(axFPGABitStream_red));
+	printk("Size of the bitstream (green): %d\n", sizeof(axFPGABitStream_green));
+
+
```
Comment (tgorochowik) [inline samples/drivers/fpga/fpga_controller/src/main.c:33]: I don't think we need these whitespaces here

**Review by tgorochowik — COMMENTED** [review]
Looks reasonable, thank you for working on this!

One minor nitpick

**Review by MaureenHelm — CHANGES_REQUESTED** [review]

File: samples/drivers/fpga/fpga_controller/README.rst [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+
+For the QuickLogic QuickFeather board:
+```bash
+west build -b quick_feather samples/drivers/fpga/fpga_controller
+```
+
+For FPGA Controller shell application:
+
+```bash
+west build -b quick_feather samples/drivers/fpga/fpga_controller -DCONF_FILE=prj_shell.conf
```
[the remark this answers is not in the document]
Comment (aasthagr) [reply samples/drivers/fpga/fpga_controller/README.rst]: Done. Thanks!!

File: samples/drivers/fpga/fpga_controller/src/main.c (hunk 2 above)
Comment (aasthagr) [reply samples/drivers/fpga/fpga_controller/src/main.c]: Yes, Removed them.

**Review by dcpleung — APPROVED** [review]

**Review by tgorochowik — APPROVED** [review]

**Review by chen-png — APPROVED** [review]

**Review by MaureenHelm — APPROVED** [review]

**Pull request merged**